### ISSUE TITLE
Darc rules

### ISSUE THREAD

**Issue by ineiti** [dedis/cothority#1605]
This is a collective issue about darcs.

1. Currently if a darc has the `evolve` rule set, the holder of the private key can evolve the darc any way he wants, adding rules to `spawn`, `invoke` or `delete`. There should be a way to restrict how a darc can evolve or what kind of darcs can be spawned
2. The `invoke`-rule of the darcs only holds the method-name, but not the name of the contract. We definitely need to move from `invoke:evolve` to `invoke:darc_evolve` or whatever syntax we chose, but we need to include the contract-name in the `invoke` command.

**Comment by ineiti** [comment]
For 1., see #1625

**Comment by kc1212** [comment]
Regarding 2, should we consider making the rules a json string? like:
```js
{
  "action": "invoke",
  "contract": "value",
  "command": "update_config"
}
```
we could even take a step further by adding the expression in there too, e.g.,
```
{
  "action": {
    "action": "invoke",
    "contract": "value",
    "command": "update_config"
  },
  "expression": "key1 || key2 || key3"
}
```

An advantage of using json or some other standardised thing is that we don't need to change the code a lot in the future in case we want to change the rules again in the future since the parsing is the same. But we still need to consider the timing of v3, using json in darcs will involve a lot of work.

**Comment by kc1212** [comment]
Closing because 1 is described in https://github.com/dedis/cothority/issues/1625 and 2 is resolved in #1673
